### PR TITLE
CORTX-30326:Hybrid API framework improvement for all the capacity API's

### PR DESCRIPTION
**Pull request by Pranali04796**
Signed-off-by: Pranali04796 <pranali.ugale@seagate.com>

# Problem Statement
- JIRA id: [CORTX-30326](https://jts.seagate.com/browse/CORTX-30326)
- Hybrid API framework improvement for all the capacity API's
# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM
- Build Details:
o1
- Deployment details:
o2
- Add API in hybrid api's:

1. Add decorator @csmauth.hybrid
hybrid
2. Add api in hybrid api's list
hybrid api

Test case1: conf key value for hyrbid api is **disable**
o4 conf key value disable
Expected result: It should work without bearer token:
o4 conf key value disable api becomes public
Test case1: conf key value for hyrbid api is **enable**
o5 conf key with auth enable
Expected result: It should not work without bearer token:
o5 conf key enable token not provided

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: csm/core/agent/api.py [hunk 1]
```diff
@@ -292,10 +292,9 @@ async def session_middleware(request, handler):
         is_public = await CsmRestApi._is_public(request)
         is_hybrid = await CsmRestApi._is_hybrid(request)
         if is_hybrid:
+            is_public = True
             conf_key = CsmRestApi._retrieve_config(request)
-            if conf_key == "disable" or conf_key == "Disable":
-                is_public = True
-            else:
+            if conf_key == "enable" or conf_key == "Enable":
```
Comment (pawankumarsrivastava) [inline csm/core/agent/api.py:297]: enable/Enable keep this in const file. I

File: csm/core/controllers/view.py [hunk 2]
```diff
@@ -35,7 +35,8 @@ class CsmAuth:
     ATTR_PERMISSIONS = '_csm_auth_permissions_'
     # ADD all hybrid api's here with required keys, use hybrid decorator
     # TODO: make it dynamic by adding required key as param to hybrid decorator.
-    HYBRID_APIS = {"GET:/api/v2/metrics/stats/perf": const.AUTH}
+    HYBRID_APIS = {"GET:/api/v2/metrics/stats/perf": const.AUTH,
+                   "GET:/api/v2/capacity/status/bytecount": const.AUTH}
```
Comment (pawankumarsrivastava) [inline csm/core/controllers/view.py:39]: Do not add bytecount API here. We have not informed QA about it. Test it thoroughly.

This change will make bytecount api public

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]

File: csm/core/agent/api.py (hunk 1 above)
Comment (Pranali04796) [reply csm/core/agent/api.py]: Added const as per comment

File: csm/core/controllers/view.py (hunk 2 above)
Comment (Pranali04796) [reply csm/core/controllers/view.py]: Removed bytecount api from hybrid api's. Will add it for testing

**Review by pawankumarsrivastava — APPROVED** [review]

**Review by rohitkolapkar — APPROVED** [review]

**Pull request merged**